### DISCUSSION TITLE
Question regarding the CC2351, device-limit and device removal

### DISCUSSION THREAD

**Discussion by abraxas767** [Koenkk/zigbee2mqtt#5577] [General]
Hi there,

My situation:

I got 25 Aquara Temperature Sensors which i want to connect with the CC2351. I reach the device limit at the 20th sensor. Now I want to get rid of one of the connected sensors in the network and pair another one. If I remove / force remove (via mqtt) a sensor, the device limit is still present and I cant connect other sensors even though I now have only 19 devices connected.

I get that battery powered devices cant receive removal requests, but is this really the default behaviour of the CC2351. If I press the reset button to remove a sensor, the behaviour is as I wished it to be. But do I really have to press 20 reset-buttons if I truly want to get rid of them? Is there an easier way?

I hope my question makes sense :/

Thanks for helping out :)

regards

**Comment by Koenkk** [comment]
If you remove devices via force remove the only way to free up this place in the device table is by reflashing the CC2531 (won't require repairing of devices).

**Reply by abraxas767** [reply]
Thank you very much for your reply and clearing things up! 
I really appreciate your work.
